### DISCUSSION TITLE
Stopped working after 12ish hours

### DISCUSSION THREAD

**Discussion by GBurchel** [cmcgerty/Sofar2mqtt#32] [Q&A]
I had it working and pulling in the data into home assistant, but after about 12 hours it stopped and now just shows the crc fault error with the 485 error popping up every so often? 
Any thoughts as to what's happened

**Comment by headcloudmonkey** [comment]
I would get that on occasions. I ended up bringing my ME3000 inverter firmware up to 2.70 and adding a 120ohm resistor across the RS485 connector that plugs into the inverter. Do you see  if the RS485 icon disappears from the inverter display? What inverter is it?

**Reply by OGB1952** [reply]
Thanks for the termination resistor tip!  I've been getting random CRC errors and have tried replacing the RS485 board, fitting screened twisted pair cable (both made slight improvements), but adding a terminating resistor across the ME3000SP end seems to have fixed things.  I can only imagine that the RS485 port on the inverter may not be properly terminated internally.  

FWIW, the same applies to some of the RS485 modules around.  I'm using a Waveshare MAX3485 module and can confirm that this does have the correct 120 ohm terminating resistor between A and B, but some of the no-name boards don't.

**Reply by Hein1966** [reply]
Indeed thank you for this solution. Truly amazing.  thank you very much.

**Reply by Banana5kin** [reply]
I too am having an issue where the inverter RS485 icon disappears and the only way to get things working again is by restarting the inverter.  I'd say this 120hom resister fix would help me.  Does the 120ohm resister go across the A & B pins?  Many thanks.

**Reply by OGB1952** [reply]
I just fitted it between the A and B pins on the RS485 connector on the inverter.   Seems to have fixed the CRC errors for me.  Other issue I had was finding a decent RS485 to serial module.  There are now red ones around that won't work at 3.3V.  I used a Waveshare module that definitely has a genuine MAX3485.  I suspect that a fair few of the modules around may have fake chips, and may be marginal when run at 3.3V.

**Comment by cliffdude** [comment]
I have the hyd-4000es and have not had issues with rs485, except for the
odd time when the inverter simply stops talking to the board. I then reset
everything ajd all is good for another month or 2.

A terça, 21/03/2023, 10:32, OGB1952 ***@***.***> escreveu:

**Comment by cliffdude** [comment]
I have a few blue (no name) aliexpress modules that should be identical, 2
work perfectly and the other 2 don't! You can never be sure that the cheap
modules are 100%
My advice is get 2 slightly different models and try both, or pay up for
the brand names.

**Reply by OGB1952** [reply]
I've now tried three Waveshare modules (two are intended for other projects) and all three work perfectly, even without the additional terminating resistor.  Not much more expensive than the Aliexpress modules ($4 plus postage), plus they have the option of using an RJ11 connector: https://www.waveshare.com/rs485-board-3.3v.htm

I fitted the Waveshare module so the RJ11pokes through a hole cut in a project box.  I re-purposed a spare phone cable to make a neat way to connect to the inverter:

![Waveshare connector](https://user-images.githubusercontent.com/110398037/235888809-86f2e385-1037-4767-b92e-98fc13e2ab43.jpg)

**Comment by Banana5kin** [comment]
Nice find.  Just ordered 2 of these, quite reasonable around $13 including registered airmail.  

I noticed the other day despite having the 120 ohm resistor fitted the max3485 didn't crap out totally it was receiving incomplete data from the inverter.  The RS485 symbol on the inverter was still illuminated but randomly data was dropped.  Restarting the Wemos D1 did the trick, didn't need to restart the inverter as was the case previously.  We'll see how these new modules work.

⁣Get BlueMail for Android ​
